pkg/apis/sources/v1alpha2/apiserver_conversion_test.go

```diff
@@ -32,3 +40,294 @@ func TestApiServerConversionBadType(t *testing.T) {
 		t.Errorf("ConvertFrom() = %#v, wanted error", good)
 	}
 }
+
+func TestApiServerSourceConversionRoundTripUp(t *testing.T) {
+	// Just one for now, just adding the for loop for ease of future changes.
+	versions := []apis.Convertible{&v1beta1.ApiServerSource{}}
+
+	path, _ := apis.ParseURL("/path")
+	sink := duckv1.Destination{
+		Ref: &duckv1.KReference{
+			Kind:       "Foo",
+			Namespace:  "Bar",
+			Name:       "Baz",
+			APIVersion: "Baf",
+		},
+		URI: path,
+	}
+	sinkUri, _ := apis.ParseURL("http://example.com/path")
```

Comment:
You can use `apis.HTTP()` here

Reply:
`apis.ParseURL` is used in a lot of places in the codebase. I will just keep it for now. We can change all of them in another PR.

Reply:
another clean up issue created https://github.com/knative/eventing/issues/3623